src/main/scala/io/iohk/ethereum/vm/Program.scala

```diff
@@ -16,6 +18,30 @@ case class Program(code: ByteString) {
   def getBytes(from: Int, size: Int): ByteString =
     code.slice(from, from + size).padTo(size, 0.toByte)
 
+  lazy val validJumpDestinations: Seq[Int] = validJumpDestinationsAfterPosition(0)
```

Comment:
If anything that should be a `Set`. But what is the advantage of keeping those in a separate collection?

The `code` is already an `IndexedSeq`, so verifying whether a position is a valid JUMPDEST should be as easy as:

```
def isValidJumpDest(pos: Int): Boolean =
  getByte(pos).contains(JUMPDEST.code)
```

and just as efficient (if not more) as looking it up in precomputed `Set`.